conda_forge_tick/utils.py

```diff
@@ -517,30 +527,145 @@ def parse_recipe_yaml_local(
         for some errors. Have fun.
     """
 
-    return _render_recipe_yaml(text)[0]["recipe"]
+    rendered_recipe = _render_recipe_yaml(text)[0]["recipe"]
 
 
-def _render_recipe_yaml(text: str) -> dict[str, Any]:
+def _render_recipe_yaml(
+    text: str,
+    cbc_path=None,
+) -> list[dict[str, Any]]:
     """
     Renders the given recipe YAML text using the `rattler-build` command-line tool.
 
-    Args:
-        text (str): The recipe YAML text to render.
+    Parameters
+    ----------
+    text : str
+        The recipe YAML text to render.
+    cbc_path : str, optional
+        The path to global pinning file.
 
-    Returns:
-        dict[str, Any]: The rendered recipe as a dictionary.
+    Returns
+    -------
+    dict[str, Any]
+        The rendered recipe as a dictionary.
     """
+    variant_config_flags = [] if cbc_path is None else ["--variant-config", cbc_path]
     res = subprocess.run(
-        ["rattler-build", "build", "--render-only"],
+        ["rattler-build", "build", "--render-only"] + variant_config_flags,
         stdout=subprocess.PIPE,
-        stderr=subprocess.DEVNULL,
         text=True,
         input=text,
         check=True,
     )
     return json.loads(res.stdout)
 
 
+def _validate_parsed_recipes(
+    parsed_recipes: list[dict[str, Any]],
+) -> list[SimpleRecipe]:
+    return [SimpleRecipe.model_validate(recipe["recipe"]) for recipe in parsed_recipes]
+
+
+def _parse_validated_recipes(
```

Comment:
To make this robust validation errors, we likely need to rewrite this code to work on the recipe as a dictionary and not as a model object.

Reply:
Yeah, will do so.